semantic-ast/docs/json-ASTs.md

```diff
@@ -0,0 +1,42 @@
+**⚠️WIP:⚠️** _I'm working through the open questions listed below and will move them up into the Design section as I answer them._
+
+# JSON output for Python
+
+`semantic-ast` is dedicated to the production of ASTs. The package provides a CLI for auto-generating ASTs from source code. This is the output of the `parseByteString` utility in `haskell-tree-sitter`. Currently, we only parse Python source, but this will soon change as we add languages. In addition to supporting more languages, consumers of this library should also be able to specify the output format in which they'd like to display generated ASTs. JSON is one of such formats, and one we can prioritize given its standard usage across a variety of applications and use-cases.
```

Comment:
I'd like to argue for just including this in semantic: specifically the existing `semantic parse` command and the APIs that are surfaced to semanticd. I think it's advantageous for the existing CLI to just get better AST support (more languages, precise ASTs, faster parsing) instead of expecting consumers to go find and use another utility.

Reply:
I was working off the vision @robrix outlined for semantic [here](https://github.com/github/semantic-code/blob/321ae2d13c7a01eee0470fde1f475e713838f7e0/architecture/README.md) in his architecture diagram and also in ongoing conversations for how he'd like us to move toward eventually deprecating the existing CLI altogether. 

I think the goals of `semantic-ast` were never really clarified anywhere though. I think discussing the future of `Semantic` and whether we want to move toward this world of smaller packages or not, and to what extent this is part of that transition probably reflects whether we subsume this stuff into Semantic's existing CLI.

Reply:
That doc just talks about splitting up the implementation into different packages, right? My reading of it has always been that there would still be a single top-level `semantic` package, that would primarily pull in all of the lower-level implementation packages and provide the `semantic` CLI program. That would still be the primary “UI” to access all of our functionality.  If there are any programs in any of the lower packages, they would not be installed, and would only be used for testing purposes.